lib/consul_application_settings/settings_provider.rb

```diff
@@ -0,0 +1,16 @@
+module ConsulApplicationSettings
+  # Provides access to settings stored in Consul or in file system
+  class SettingsProvider
+    def initialize(base_path, config)
+      @consul_provider = ConsulProvider.new(base_path, config)
+      @file_provider = FileProvider.new(base_path, config)
+    end
+
+    def get(path)
+      consul_value = @consul_provider.get(path)
+      !consul_value.nil? && consul_value != '' ? consul_value : @file_provider.get(path)
```

Comment:
Should we think about the case when I want to clear (set to nil) the particular setting using consul?

Reply:
Good point. Didn't think of that. Let's leave it until we run into this situation?